chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/RoomStore.kt

```diff
@@ -48,79 +46,119 @@ internal class RoomStore(
             event: UserSubscriptionEvent
     ): List<UserSubscriptionEvent> =
             when (event) {
+                // TODO: extract well named funs with clear responsibility
                 is UserSubscriptionEvent.InitialState -> {
                     val knownRooms = this.toList()
-                    val removedFrom = knownRooms.filterNot {
-                        event.rooms.contains(it)
-                    }.onEach {
-                        this -= it
-                    }.map {
-                        UserSubscriptionEvent.RemovedFromRoomEvent(it.id)
-                    }
 
                     val addedTo = event.rooms.filterNot {
                         knownRooms.contains(it)
                     }.onEach {
                         this += it
+                    }.map { room ->
+                        UserSubscriptionEvent.AddedToRoomEvent(room,
+                                event.readStates.find { it.roomId == room.id }!!,
+                                event.memberships.find { it.roomId == room.id }!!)
+                    }
```

Comment:
It concerns me that we supply the readStates and memberships here, but not where we `+=` the room in to the store.

I'd really like to separate the storage of the three different entities (rooms, memberships and read states) within the `RoomStore` and then mix them together whenever we retrieve a Room.

Reply:
Having them returned separately from all endpoints in the backend (consistent with how they come on the subscription) would *really* help with this...

Reply:
I realise that the `event.rooms` have been combined correctly, but it's quite difficult to follow and be sure about.